wcfsetup/install/files/lib/system/moderation/queue/AbstractModerationQueueManager.class.php

```diff
@@ -119,6 +120,112 @@ protected function addEntry($objectTypeID, $objectID, $containerID = 0, array $a
 		ModerationQueueManager::getInstance()->resetModerationCount();
 	}
 	
+	/**
+	 * Adds multiple entries to moderation queue at once.
+	 *
+	 * In contrast to `addModeratedContent()`, this method expects the container ids to be
+	 * passed as a parameter. If no container id is given for a specific object id, `0` is
+	 * used as container id.
+	 *
+	 * This method is intended for bulk processing.
+	 * 
+	 * @param	integer		$objectTypeID
+	 * @param	integer[]	$objectIDs
+	 * @param	integer[]	$containerIDs		format: `objectID => containerID`
+	 * @param 	array		$additionalData
+	 */
+	protected function addEntries($objectTypeID, array $objectIDs, array $containerIDs, array $additionalData = []) {
+		$conditionBuilder = new PreparedStatementConditionBuilder();
+		$conditionBuilder->add('objectTypeID = ?', [$objectTypeID]);
+		$conditionBuilder->add('objectID IN (?)', [$objectIDs]);
+		
+		$sql = "SELECT	queueID, objectID
+			FROM	wcf".WCF_N."_moderation_queue
+			" . $conditionBuilder;
+		$statement = WCF::getDB()->prepareStatement($sql);
+		$statement->execute($conditionBuilder->getParameters());
+		$existingQueueIDs = $statement->fetchMap('objectID', 'queueID');
+		
+		// create new moderation queue entries for new objects
+		$newObjectIDs = array_diff($objectIDs, array_keys($existingQueueIDs));
+		
+		$itemsPerLoop = 1000;
+		$batchCount = ceil(count($newObjectIDs) / $itemsPerLoop);
+		$userID = WCF::getUser()->userID ?: null;
+		$serializedData = serialize($additionalData);
+		
+		WCF::getDB()->beginTransaction();
+		for ($i = 0; $i < $batchCount; $i++) {
+			$batchObjectIDs = array_slice($newObjectIDs, $i * $itemsPerLoop, $itemsPerLoop);
+			
+			$parameters = [];
+			foreach ($batchObjectIDs as $objectID) {
+				$parameters = array_merge($parameters, [
+					$objectTypeID,
+					$objectID,
+					$containerIDs[$objectID] ?? 0,
+					$userID,
+					TIME_NOW,
+					TIME_NOW,
+					$serializedData
+				]);
+			}
+			
+			$sql = "INSERT INTO	wcf" . WCF_N . "_moderation_queue
+						(objectTypeID, objectID, containerID, userID, time, lastChangeTime, additionalData)
+				VALUES		(?, ?, ?, ?, ?, ?, ?)" . str_repeat(', (?, ?, ?, ?, ?, ?, ?)', count($batchObjectIDs) - 1);
+			$statement = WCF::getDB()->prepareStatement($sql);
+			$statement->execute($parameters);
+		}
+		WCF::getDB()->commitTransaction();
+		
+		// update existing moderation queue entries
```

Comment:
Is this comment correct here?

Reply:
Yes, see the `UPDATE` query below.

Reply:
It's strange to see freestanding comments (i.e. with an empty line afterwards).